### DISCUSSION TITLE
How to mkdocs material can process  from multiple repo and sync

### DISCUSSION THREAD

**Discussion by SavantEVO** [squidfunk/mkdocs-material#6366] [Q&A]
Hello everyone! I need help 

> **How to mkdocs material process get file README.md from multiple repo and can be synced?'**

Please help me or give me some advice or something to process this. Moreover, If get README.md from another Repo, and can be synced but if the repo Gitlab (private needs to username, and password) I saw have option with an access token to clone the repo. But I can not use this option can solve this problem

**Comment by alexvoss** [comment]
Hope I understand this correctly. What you need to do is to inclue README files from other repos that need authentication. Use of a token does sound like the solution of choice for anything that needs to run automatically, such in a pipeline. It is usually that, depositing the username and password somewhere (less ideal) or using SSH keys. That tends to be the same for GitHub, Gitlab and other systems. You may want to consult the Gitlab docu and ask in a Gitlab-related forum if you have questions about this.

If you are building locally, could you simply clone the other repos and pull copy their READMEs before building? Ideally, this would be scripted but with a bit of diligence can be done manually as well.

**Reply by SavantEVO** [reply]
So if I write a script Python with behavior clones repo Gitlab optional (URLrepo; GET: README.md; SET time to update file (README.md). How can I embed in mkdocs ?. Thanks you

**Reply by alexvoss** [reply]
You could try to write is as a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks). I would suggest attaching it to the [on_pre_build](https://www.mkdocs.org/dev-guide/plugins/#on_pre_build) event.

**Reply by SavantEVO** [reply]
Can you help me after running [on_pre_build]? I saw that this will be a cloned repo run by myscript.py. But I do not know how mkdocs can know to reprocess and update when I update something in the file README.md.

**Reply by alexvoss** [reply]
You may want to try the [watch setting](https://www.mkdocs.org/user-guide/configuration/#watch). There is a note in the documentation that the paths provided on the command-line are not relative to the configuration file. That might be the way to go. Another approach would be to use symbolic links to bring the README.md files into your project.

**Reply by SavantEVO** [reply]
@alexvoss I have done the script update README.md when the repo Gitlab updates with behavior (commit, merge) the file READMEmd will be added in folders docs. That behavior will be fine or maybe have some problem. Please give me advice thank you so much!

**Reply by alexvoss** [reply]
I am afraid I don't understand what you are asking - there is no information in your post to go on.
